plugins/inputs/syslog/syslog.go

```diff
@@ -279,20 +280,51 @@ func (s *Syslog) handle(conn net.Conn, acc telegraf.Accumulator) {
 		conn.Close()
 	}()
 
-	if s.ReadTimeout != nil && s.ReadTimeout.Duration > 0 {
-		conn.SetReadDeadline(time.Now().Add(s.ReadTimeout.Duration))
-	}
+	for {
+		data := &bytes.Buffer{}
 
-	var p *rfc5425.Parser
-	if s.BestEffort {
-		p = rfc5425.NewParser(conn, rfc5425.WithBestEffort())
-	} else {
-		p = rfc5425.NewParser(conn)
-	}
+		// read the data
+		if s.ReadTimeout != nil && s.ReadTimeout.Duration > 0 {
+			conn.SetReadDeadline(time.Now().Add(s.ReadTimeout.Duration))
+		}
 
-	p.ParseExecuting(func(r *rfc5425.Result) {
-		s.store(*r, acc)
-	})
+		n, err := io.Copy(data, conn)
```

Comment:
Won't this read until EOF?  How many messages can syslog send over a single connection.  If it allows persistent connection then this could be very large.

Reply:
A receiver MUST use the message length to delimit a syslog messages, not `EOF`.
Ie., 14 octets for the following message `<1>1 - - - - -`, thus `14 <1>1 - - - - -`.

RFC5425 specifies only that receivers SHOULD be able to process messages with lengths up to and including 8192 octets.

There is not maximum.